### PR TITLE
Show phone location on map

### PR DESCRIPTION
**Pull request by Jorque**
Asks user for permission and show his location on the map. If the user does not give his permission, he can still use the map but he will be asked for permission each time he wants accesses the map (opens ItineraryCreateActivity). I could not test the permission asking popup as it seems that espresso has no way to interact with it.

### CODE REVIEW

**Review by Anghul — DISMISSED** [review]

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/MapActivityTest.kt [hunk 1]
```diff
@@ -46,11 +58,17 @@ class MapActivityTest {
...
     }
 
-    @BindValue val authService: AuthenticationService = mock(AuthenticationService::class.java)
+    @BindValue
+    val authService: AuthenticationService = mock(AuthenticationService::class.java)
 
     @Before
     fun setUp() {
         Intents.init()
+        `when`(context.checkSelfPermission(ACCESS_FINE_LOCATION))
```
Comment (jonasblanc) [inline app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/MapActivityTest.kt:67]: Is it necessary to have it here ? If you redefine it for every test ?

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/MapActivityTest.kt [hunk 2]
```diff
@@ -72,4 +90,37 @@ class MapActivityTest {
...
+                0,
+                arrayOf("android.permission.ACCESS_FINE_LOCATION"),
+                intArrayOf(-1)
+            )
+        }
+
+    }
+
+    @Test
+    fun allowLocatePositionWorks() {
```
Comment (jonasblanc) [inline app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/MapActivityTest.kt:112]: I'm not sure what those tests are testing. Can you explain it to me please ?

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/MapActivityTest.kt (hunk 1 above)
Comment (Jorque) [reply app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/MapActivityTest.kt:67]: No, I've just left it here when I was trying some stuff. I will delete it.

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/MapActivityTest.kt (hunk 2 above)
Comment (Jorque) [reply app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/MapActivityTest.kt:112]: It does not test anything yet. For the moment it is just here to increase coverage but I'm trying to figure out a way to make it test something.

**Review by loh0um — APPROVED** [review]

**Pull request merged**